src/playlist-selectors.js

```diff
@@ -124,27 +146,32 @@ const simpleSelector = function(master, bandwidth, width, height) {
   let resolutionPlusOneSmallest = [];
   let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  stableSort(sortedPlaylists, comparePlaylistBandwidth.bind(null, useAverageBandwidth));
 
   // filter out any playlists that have been excluded due to
   // incompatible configurations or playback errors
   sortedPlaylists = sortedPlaylists.filter(Playlist.isEnabled);
   // filter out any variant that has greater effective bitrate
   // than the current estimated bandwidth
   bandwidthPlaylists = sortedPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.BANDWIDTH &&
-           elem.attributes.BANDWIDTH * Config.BANDWIDTH_VARIANCE < bandwidth;
+    const elemBandwidth = getBandwidth(elem, useAverageBandwidth);
```

Comment:
`For sorting, if a playlist has a higher bandwidth, then it should have a higher average-bandwidth as well`
That is actually part of the reason why AVERAGE-BANDWIDTH can sometimes make a difference, is because it is possible for a playlist to have a higher BANDWIDTH but a lower AVERAGE-BANDWIDTH than another playlist (and vice versa).

`Since AVERAGE-BANDWIDTH is an optional property, if we opt to use it in either a sorting or filtering scenario, all it takes is for one stream in the list to not include the attribute and all the sort/filters will be incorrect.`
In practice, most m3u8s should exhibit all or nothing, either AVERAGE-BANDWIDTH is present on all playlists, or it is present on none. However, in the event that it is only available on some, AVERAGE-BANDWIDTH will be used unless it is not available, at which point the compared to playlist will use BANDWIDTH. While that isn't an "apples-to-apples" comparison,  it should be using a "best available information" approach.